tidb: update tidb_index_serial_scan_concurrency docs (#21938)
pingcap/docs: statistics.md

#### `tidb_build_stats_concurrency`

> Used to control the concurrency of table or partition analysis — for example, the number of partitions or table tasks that can be processed simultaneously. The default value is `2`. The default value is `4` for v7.4.0 and earlier versions.

Reviewer: ```suggestion
This variable controls the concurrency of table or partition analysis (for example, the number of partitions or table tasks that can be processed simultaneously). The default value is `2`. The default value is `4` for v7.4.0 and earlier versions.
```

#### `tidb_build_sampling_stats_concurrency`

1. Used to control the concurrency level for merging samples collected from different regions.  

> 2. Used to control the concurrency for special indexes (such as indexes on generated virtual columns), for example, the number of special indexes that can collect statistics simultaneously.

Reviewer: ```suggestion
This variable controls the following aspects of `ANALYZE` concurrency:

- The concurrency for merging samples collected from different regions.
- The concurrency for special indexes (such as indexes on generated virtual columns), for example, the number of indexes that TiDB can concurrently collect statistics for.
```

The default value is `2`.

#### `tidb_analyze_partition_concurrency`

> Used to control the concurrency for saving analyze results (writing TopN and histograms to system tables). The default value is `2`. The default value is `1` for v7.4.0 and earlier versions.

Reviewer: ```suggestion
This variable controls the concurrency for saving analyze results (writing TopN and histograms to system tables). The default value is `2`. The default value is `1` for v7.4.0 and earlier versions.
```

> 

Reviewer: Does `tidb_analyze_partition_concurrency` only take effect on partitioned tables, as "partition" is included in the variable name?

Author: No, it also affects other non-partitioned tables. The name is poor and confusing. We plan to merge and deprecate this in the near future.

#### `tidb_analyze_distsql_scan_concurrency`

1. Used to control the concurrency of scanning TiKV regions.  

> 2. Used to control the concurrency of scanning regions for special indexes (indexes generated from virtual columns).

Reviewer: ```suggestion
This variable controls the following aspects of `ANALYZE` concurrency:

- The concurrency of scanning TiKV regions. 
- The concurrency of scanning regions for special indexes (indexes generated from virtual columns).
```

#### `tidb_build_stats_concurrency`

> This variable controls the concurrency of table or partition analysis (for example, the number of partitions or table tasks that can be processed simultaneously). The default value is `2`. The default value is `4` for v7.4.0 and earlier versions.

Reviewer: ```suggestion
This variable controls the concurrency for building statistics during manual `ANALYZE`, such as the number of table or partition analysis tasks that can be processed simultaneously. The default value is `2`. The default value is `4` for v7.4.0 and earlier versions.
```

#### `tidb_build_sampling_stats_concurrency`

This variable controls the following aspects of `ANALYZE` concurrency:

- The concurrency for merging samples collected from different regions.
- The concurrency for special indexes (such as indexes on generated virtual columns), for example, the number of indexes that TiDB can concurrently collect statistics for.

The default value is `2`.

#### `tidb_analyze_partition_concurrency`

> This variable controls the concurrency for saving analyze results (writing TopN and histograms to system tables). The default value is `2`. The default value is `1` for v7.4.0 and earlier versions.

Reviewer: ```suggestion
This variable controls the concurrency for saving `ANALYZE` results (writing TopN and histograms to system tables). The default value is `2`. The default value is `1` for v7.4.0 and earlier versions.
```

#### `tidb_build_stats_concurrency`

This variable controls the concurrency for building statistics during manual `ANALYZE`, such as the number of table or partition analysis tasks that can be processed simultaneously. The default value is `2`. The default value is `4` for v7.4.0 and earlier versions.

#### `tidb_build_sampling_stats_concurrency`

This variable controls the following aspects of `ANALYZE` concurrency:

> - The concurrency for merging samples collected from different regions.

Reviewer: ```suggestion
- The concurrency for merging samples collected from different Regions.
```

> - The concurrency for special indexes (such as indexes on generated virtual columns), for example, the number of indexes that TiDB can concurrently collect statistics for.

Reviewer: ```suggestion
- The concurrency for collecting statistics on special indexes (such as indexes on generated virtual columns), for example, the number of indexes that TiDB can concurrently collect statistics for.
```

The default value is `2`.

#### `tidb_analyze_partition_concurrency`

This variable controls the concurrency for saving `ANALYZE` results (writing TopN and histograms to system tables). The default value is `2`. The default value is `1` for v7.4.0 and earlier versions.

#### `tidb_analyze_distsql_scan_concurrency`

This variable controls the following aspects of `ANALYZE` concurrency:

- The concurrency of scanning TiKV regions. 

> - The concurrency of scanning regions for special indexes (indexes generated from virtual columns).

Reviewer: ```suggestion
- The concurrency of scanning TiKV Regions. 
- The concurrency of scanning Regions for special indexes (indexes generated from virtual columns).
```
